W-14476576 DC sync not req duke (includes W-14476577,  W-14476579,  W-14776338) (#263)
mulesoft/docs-code-builder: modules/ROOT/pages/des-create-api-specs.adoc

include::reuse::partial$beta-banner.adoc[tag="anypoint-code-builder"]

Use Anypoint Code Builder to create and test API specs before publishing to Anypoint Exchange:

. <<design-new-api-spec, Create a new API specification project>> or <<import-spec, import an API specification from Design Center>>.
. <<open-output-panel, Track progress in the output panel>>.
. <<review-spec-console, Review your spec in the API console>>.
. <<test-spec, Test your spec using the mocking service>>.

Use the <<example-oas-api-spec,`oas-example`>> API specification to explore the functionality of Anypoint Code Builder. 

== Before You Begin

* xref:start-acb.adoc[Set up and access the web or desktop IDE].
* Understand the basics of designing API specs. 
+
See the xref:tutorials.adoc[] for details.
* Have some familiarity with xref:access-management::business-groups.adoc[business groups].
+

> When you publish your API fragment to Exchange from Anypoint Code Builder, the IDE requests the name of the business group.

Author: ```suggestion
When you publish your API spec to Exchange from Anypoint Code Builder, the IDE requests the name of the business group. See xref:access-management::business-groups.adoc[].
```

Author: make sure xref:start-scm.adoc exists

| *Project Name* | Unique name for your project.

> This name is used as the API spec title in Exchange and name of the spec file.

Reviewer: It is also used for the directory being created for the project

Author: done

> This name is used as the API spec title in Exchange and name of the spec file.

Author: ```suggestion
This name is used as the API spec title in Exchange, the name of the spec file, and the name of the project's root directory.
```

| *API Specification Language* | See xref:des-designing-api-specs.adoc#supported-spec-languages[Supported OAS and RAML Versions].

Select *OAS 3.0 (YAML)* to use the <<example-oas-api-spec,example>> in this procedure.

> | *Business Group* | Business Group in Anypoint Platform for your API specification.

Author: This was removed from the creation form.

| *API Specification Language* | See xref:des-designing-api-specs.adoc#supported-spec-languages[Supported OAS and RAML Versions].

Select *OAS 3.0 (YAML)* to use the <<example-oas-api-spec,example>> in this procedure.
|===
. Click *Create Project*.
+
If prompted, trust the authors of the files in the folder.
+
When the project is ready for editing, the API project opens the spec in the Editor view, for example, this OAS 3.0 (YAML) spec:
+
image::des-oas-create.png[New OAS 3.0 (YAML) API project]
. Continue designing your API spec. 
+
As you enter elements, use xref:start-discover-ui.adoc#auto-complete[auto-complete] (Ctrl+Space) to display available options within the context. 

[[import-spec]]
== Import an API Specification from Design Center

> Import an API spec from Design Center, and optionally, keep your project in sync with Design Center by using the Anypoint source control management (SCM) system. Alternatively, use Git or another SCM that is compatible with VS Code. For more information, see xref:des-designing-api-specs.adoc#anypoint-scm[Source Control for API Design Projects].

Reviewer 2: Only VS Code is mentioned. Is this synonymous with Desktop IDE? What are the implications for Cloud IDE? Is Git SCM only compatible with Desktop VS Code? I assume so, but can you somehow use Git in Cloud IDE too? Is this mentioned somewhere and I missed it? If it's not mentioned, newbie customers might have similar questions.

Author: Well, hmm, I see what you mean. Technically, the cloud IDE is an implementation of VS Code in the cloud. You can add extensions to the cloud IDE. This is a VS Code compatibility thing though. I can add a bit more to clarify.

Author: I was able to install the extensions for SVN and Perforce into my cloud IDE instance, so I'll just reword a bit.

Author: done

> Import an API spec from Design Center, and optionally, keep your project in sync with Design Center by using the Anypoint source control management (SCM) system. Alternatively, use Git or another SCM that is compatible with VS Code. For more information, see xref:des-designing-api-specs.adoc#anypoint-scm[Source Control for API Design Projects].

Author: ```suggestion
Import an API spec from Design Center, and optionally, keep your project in sync with Design Center by using the Anypoint source control management (SCM) system. Alternatively, use Git or another SCM that is compatible with VS Code in the desktop or cloud IDE. For more information, see xref:des-designing-api-specs.adoc#anypoint-scm[Source Control for API Design Projects].
```

Author: The cloud IDE is build on an implementation of VS Code called code-server. That info is in the overview.

// Pointer to Command Palette
include::partial$acb-reusable-steps.adoc[tags="open-command-palette"]
. Provide the following command:
+
[source,command]
----
MuleSoft: Import API Specification from Design Center
----
. If prompted, log into Anypoint Platform through the IDE.
. Complete the import process:
.. Select your business group from the *Select a Business Group* popup. 
.. Search for the API spec in the *APIs from Design Center* field.
.. Navigate to or create a folder for the spec, and click *Select target folder*.
+
If you receive an error that the import failed (*Importing project _a_project_name_ has failed*), check your target folder for a project with the same name. To address a duplicate naming issue before importing, you can import the design project to a different target folder or delete the project from your IDE (see xref:des-delete-api-projects.adoc[]).

> //TODO: THIS REQ WILL GO AWAY WHEN THIS ISSUE IS FIXED:

Reviewer 3: Shouldn't this TODO be removed?

Author: done

> //TODO: THIS REQ WILL GO AWAY WHEN THIS ISSUE IS FIXED:

Author: ```suggestion
```

// Pointer to Command Palette
include::partial$acb-reusable-steps.adoc[tags="open-command-palette"]
. Provide the following command:
+
[source,command]
----
MuleSoft: Import API Specification from Design Center
----
. If prompted, log into Anypoint Platform through the IDE.
. Complete the import process:
.. Select your business group from the *Select a Business Group* popup. 
.. Search for the API spec in the *APIs from Design Center* field.
.. Navigate to or create a folder for the spec, and click *Select target folder*.
+
If you receive an error that the import failed (*Importing project _a_project_name_ has failed*), check your target folder for a project with the same name. To address a duplicate naming issue before importing, you can import the design project to a different target folder or delete the project from your IDE (see xref:des-delete-api-projects.adoc[]).
//TODO: THIS REQ WILL GO AWAY WHEN THIS ISSUE IS FIXED:
.. Click *Save* if you receive the request *Do you want to save the changes you made to exchange.json? Your changes will be lost if you don't save them.*. 
+

> This action saves project metadata in `exchange.json`. If you click *Don't Save*, the IDE will ask you if you want to save the file when you close the project folder. You can save at that time.

Reviewer: This issue was already fixed in version `4.5.3` for February release. We can remove this

Author: done

> .. Click *Save* if you receive the request *Do you want to save the changes you made to exchange.json? Your changes will be lost if you don't save them.*. 
> +
> This action saves project metadata in `exchange.json`. If you click *Don't Save*, the IDE will ask you if you want to save the file when you close the project folder. You can save at that time.

Author: ```suggestion
```
removing
